syntax = "proto3";

import "google/protobuf/struct.proto";

package osiris;

// Service de communication entre l'agent et le Hive
service AgentComms {
  // Enregistrement de l'agent et heartbeat
  rpc Register(RegistrationRequest) returns (RegistrationResponse);
  
  // Heartbeat et réception d'instructions
  rpc Heartbeat(HeartbeatRequest) returns (HeartbeatResponse);
  
  // Envoi des résultats de requête
  rpc SendQueryResults(QueryResult) returns (QueryResponse);
}

// Message de requête d'enregistrement
message RegistrationRequest {
  string agent_id = 1;
  string hostname = 2;
  string os_info = 3;
}

// Message de réponse d'enregistrement
message RegistrationResponse {
  string status = 1;
}

// Message de requête de heartbeat
message HeartbeatRequest {
  string agent_id = 1;
}

// Message de réponse de heartbeat
message HeartbeatResponse {
  string status = 1;
  HiveInstruction instruction = 2;
}

// Message d'instruction du Hive
message HiveInstruction {
  string query = 1;
  string query_id = 2;
}

// Message de résultat de requête
message QueryResult {
  string query_id = 1;
  google.protobuf.Struct result = 2;
  QuerySummary summary = 3;
}

// Message de résumé de requête
message QuerySummary {
  string query_id = 1;
  string status = 2;
}

// Message de réponse pour les résultats de requête
message QueryResponse {
  string status = 1;
} 